src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
```

Comment:
Similar to the previous function, using .some and .find within .filter and .forEach leads to nested loops.

1. Create a Map for quick mother lookup
2. Filter children
3. Calculate sum using .reduce